File: service/src/main/java/coding/challenge/core/converter/StringToAudioFileFormatConverter.java
package coding.challenge.core.converter;

import coding.challenge.core.model.AudioFileFormat;
import org.springframework.core.convert.converter.Converter;
import org.springframework.stereotype.Component;

/**
 * Converts string to {@link AudioFileFormat}.
 */
@Component
public class StringToAudioFileFormatConverter implements Converter<String, AudioFileFormat> {

    @Override
    public AudioFileFormat convert(String source) {
        return AudioFileFormat.fromValue(source);
    }

}

File: service-kt/src/main/kotlin/coding/challenge/core/converter/StringToAudioFileFormatConverter.kt
package coding.challenge.core.converter

import coding.challenge.core.model.AudioFileFormat
import org.springframework.core.convert.converter.Converter
import org.springframework.stereotype.Component

/**
 * Converts string to [AudioFileFormat].
 */
@Component
class StringToAudioFileFormatConverter : Converter<String, AudioFileFormat> {

    override fun convert(source: String): AudioFileFormat = AudioFileFormat.fromValue(source)

}
